### PR TITLE
Fix 3516: Change an emitter #ifdef _TARGET_AMD64_ to !LEGACY_BACKEND

### PR DESCRIPTION
**Pull request by BruceForstall**

### CODE REVIEW

**Comment by BruceForstall** [comment]
@sivarv @adiaaida PTAL
/cc @dotnet/jit-contrib

File: src/jit/emitxarch.cpp [hunk 1]
```diff
@@ -8794,14 +8794,15 @@ BYTE*       emitter::emitOutputCV(BYTE* dst, instrDesc* id, size_t code, CnsVal*
...
+
+#ifndef LEGACY_BACKEND
         // this instruction has a fixed size (4) src.  
         if (ins == INS_cvttss2si || ins == INS_cvtss2sd || ins == INS_vbroadcastss)
             byteSize = 4;
         // This has a fixed size (8) source.
         if (ins == INS_vbroadcastsd)
             byteSize = 8;
-#endif
+#endif // !LEGACY_BACKEND
```
Comment (sivarv) [inline src/jit/emitxarch.cpp]: I see quite a few such cases in emitxarch.cpp that needs to be enabled for RyuJIT x86.

Example:
emitInsCanOnlyWriteSSE2OrAVXReg()

Should we audit all #ifdef _TARGET_AMD64_ and fix in one shot?

Comment (BruceForstall) [inline src/jit/emitxarch.cpp]: I'll plan to do that after this one.

**Comment by sivarv** [comment]
Looks good.

**Comment by CarolEidt** [comment]
LGTM

**Comment by adiaaida** [comment]
LGTM

**Pull request merged**